### PR TITLE
[Profiler] Include more uncategorized events in memory profile

### PR DESCRIPTION
**Pull request by aaronenyeshi**
Summary: This PR adds handling for allocations / frees which we cannot prove are for Tensors. (And thus aren't assigned an ID.) These events are still important for judging overall utilization.

Test Plan: CI and Unit tests.

Differential Revision: D45458885

Pulled By: aaronenyeshi

### CODE REVIEW

**Review by anupambhatnagar — APPROVED** [review]
LGTM

File: torch/profiler/_memory_profiler.py [hunk 1]
```diff
@@ -97,7 +105,6 @@ class TensorKey:
 
     id: int
     storage: _Storage
-    device: torch.device
 
     def __repr__(self) -> str:
         return f"id={self.id}: {repr(self.storage):<24} ({self.device})"
```
Comment (davidberard98) [inline torch/profiler/_memory_profiler.py:110]: probably need to remove self.device here?

**Review by davidberard98 — APPROVED** [review]
quick comment about a stray `self.device`, otherwise LGTM!

**Review by davidberard98 — APPROVED** [review]
quick comment about a stray `self.device`, otherwise LGTM!

File: torch/profiler/_memory_profiler.py (hunk 1 above)
Comment (aaronenyeshi) [reply torch/profiler/_memory_profiler.py:110]: Since we are inheriting self.device from Key, is the better idea to make a __repr__ in class Key?

Comment (davidberard98) [reply torch/profiler/_memory_profiler.py]: ah I missed that, it looks fine then!

**Pull request closed without merging**